File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/outbound/db/newsletter/NewsletterSubscriptionJpaRepository.java
package io.miragon.example.adapter.outbound.db.newsletter;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.UUID;

public interface NewsletterSubscriptionJpaRepository extends JpaRepository<NewsletterSubscriptionEntity, UUID> {
    NewsletterSubscriptionEntity findBySubscriptionId(UUID id);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/adapter/outbound/db/newsletter/NewsletterSubscriptionJpaRepository.kt
package de.emaarco.example.adapter.outbound.db.newsletter

import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface NewsletterSubscriptionJpaRepository : JpaRepository<NewsletterSubscriptionEntity, UUID> {
    fun findBySubscriptionId(id: UUID): NewsletterSubscriptionEntity?
}